jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryResultsWindow.kt

```diff
@@ -18,20 +18,39 @@ class QueryResultsWindow(private val project: Project) : CoroutineScope by Appli
     private val toolWindow = ToolkitToolWindowManager.getInstance(project, INSIGHTS_RESULTS_TOOL_WINDOW)
     private val edtContext = getCoroutineUiContext()
 
-    fun showResults(queryDetails: QueryDetails, queryId: String, fields: List<String>) = launch {
-        val existingWindow = toolWindow.find(queryId)
-        if (existingWindow != null) {
-            withContext(edtContext) {
-                existingWindow.show()
-            }
-            return@launch
+    suspend fun showResults(queryDetails: QueryDetails, queryId: String, fields: List<String>) {
+        if (!createToolWindow(queryId)) {
+            return
         }
+
         val queryResult = QueryResultList(project, fields, queryId, queryDetails)
         withContext(edtContext) {
             toolWindow.addTab(queryId, queryResult.resultsPanel, activate = true, id = queryId, disposable = queryResult)
         }
     }
 
+    suspend fun showDetailedEvent(client: CloudWatchLogsClient, identifier: String) {
+        if (!createToolWindow(identifier)) {
+            return
+        }
+
+        val detailedLogEvent = DetailedLogRecord(project, client, identifier)
+        withContext(edtContext) {
+            toolWindow.addTab(detailedLogEvent.title, detailedLogEvent.getComponent(), activate = true, id = identifier, disposable = detailedLogEvent)
+        }
+    }
+
+    private suspend fun createToolWindow(identifier: String): Boolean {
```

Comment:
I thought `false` meant failure when I looked at how it is used, also it shows instead of creating. how about `showWindow` then inverting the logic